purchase_sale_inter_company/models/purchase_order.py

```diff
@@ -3,13 +3,20 @@
 # Copyright 2018-2019 Tecnativa - Carlos Dauden
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 
-from odoo import _, models
+from odoo import _, fields, models
 from odoo.exceptions import UserError
 
 
 class PurchaseOrder(models.Model):
     _inherit = "purchase.order"
 
+    intercompany_sale_id = fields.Many2one(
```

Comment:
I set this comment as resolve, because a PO can have several sale order (you can have several canceled so).
But I think it will be better to set the field "intercompany_sale_id" as computed (with compute_sudo and take the last created) to avoid storing a duplicated information.